### PR TITLE
ARROW-18042: [Java] Distribute Apple M1 compatible JNI libraries via mavencentral

### PR DESCRIPTION
**Pull request by rok**
This is to resolve [ARROW-18042](https://issues.apache.org/jira/browse/ARROW-18042).

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -1058,7 +1058,7 @@
                     -DCMAKE_BUILD_TYPE=Release
                     -DCMAKE_INSTALL_LIBDIR=lib
                     -DCMAKE_INSTALL_PREFIX=${arrow.dataset.jni.dist.dir}
-                    -DCMAKE_PREFIX_PATH=${project.basedir}/../java-dist
+                    -DCMAKE_PREFIX_PATH=${project.basedir}/../${os.detected.arch}/java-dist
```
Comment (rok) [inline java/pom.xml:1061]: @davisusanibar is this enough to pull the correct modules or do we need to adjust more paths?

Comment (davisusanibar) [reply java/pom.xml:1061]: Hi Rok, I could suggest that is needed to add `${os.detected.arch}` before all places working with folder `java-dist`

The you could test your changes locally with this maven command please: `mvn generate-resources -Pgenerate-libs-jni-macos-linux -N` inside of `arrow/java` folder ?

**Comment by davisusanibar** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
Closing in favor of #14472

**Pull request closed without merging**